### PR TITLE
Register Ltac2 grammar entry as "ltac2" for the Print Grammar vernacular

### LINKED ISSUE #14092: `Print Grammar ltac2` should exist
Without this, it seems quite hard to debug tactic precedences in Ltac2.

cc @ppedrot @herbelin

### PR DESCRIPTION
**Pull request by ppedrot**
Fixes #14092: Print Grammar ltac2 should exist.

- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Review by JasonGross — APPROVED** [review]
Seems to work great, thanks!

<details><summary>Current printout</summary>

```
Welcome to Coq JasonGross-X1:/home/jgross/Documents/repos/coq,pr/14093 (45e064836ec1f50300198f89c099b71d0e2059b9)
[Loading ML file ltac2_plugin.cmxs ... done]
Entry ltac2_expr is
[ "6" RIGHTA
  [ SELF; ">"; "["; q_dispatch; "]"
  | SELF; ";"; SELF
  | IDENT "now"; SELF
  | IDENT "typeclasses_eauto"; IDENT "bfs"; OPT (ltac2_expr LEVEL "0");
    OPT [ IDENT "with"; LIST1 q_ident ]
  | IDENT "typeclasses_eauto"; OPT (ltac2_expr LEVEL "0");
    OPT [ IDENT "with"; LIST1 q_ident ]
  | IDENT "eauto"; OPT (ltac2_expr LEVEL "0"); OPT (ltac2_expr LEVEL "0");
    OPT [ IDENT "using"; LIST1 constr SEP "," ];
    OPT [ IDENT "with"; q_hintdb ]
  | IDENT "new"; IDENT "auto"; OPT (ltac2_expr LEVEL "0");
    OPT [ IDENT "using"; LIST1 constr SEP "," ];
    OPT [ IDENT "with"; q_hintdb ]
  | IDENT "auto"; OPT (ltac2_expr LEVEL "0");
    OPT [ IDENT "using"; LIST1 constr SEP "," ];
    OPT [ IDENT "with"; q_hintdb ]
  | IDENT "trivial"; OPT [ IDENT "using"; LIST1 constr SEP "," ];
    OPT [ IDENT "with"; q_hintdb ]
  | IDENT "einjection"; OPT q_destruction_arg;
    OPT [ IDENT "as"; q_intropatterns ]
  | IDENT "injection"; OPT q_destruction_arg;
    OPT [ IDENT "as"; q_intropatterns ]
  | IDENT "ediscriminate"; OPT q_destruction_arg
  | IDENT "discriminate"; OPT q_destruction_arg
  | IDENT "subst"; LIST0 q_ident
  | IDENT "absurd"; constr
  | IDENT "clear"; "-"; LIST1 q_ident
  | IDENT "clear"; LIST0 q_ident
  | IDENT "revert"; LIST1 q_ident
  | IDENT "symmetry"; OPT q_clause
  | IDENT "move"; q_ident; q_move_location
  | IDENT "intro"; OPT q_ident; OPT q_move_location
  | IDENT "eexact"; constr
  | IDENT "exact"; constr
  | IDENT "erewrite"; LIST1 q_rewriting SEP ","; OPT q_clause;
    OPT [ IDENT "by"; ltac2_expr LEVEL "5" ]
  | IDENT "rewrite"; LIST1 q_rewriting SEP ","; OPT q_clause;
    OPT [ IDENT "by"; ltac2_expr LEVEL "5" ]
  | IDENT "change"; q_conversion; OPT q_clause
  | IDENT "eval"; IDENT "native_compute"; OPT [ constr; q_occurrences ];
    "in"; constr
  | IDENT "eval"; IDENT "vm_compute"; OPT [ constr; q_occurrences ]; "in";
    constr
  | IDENT "eval"; IDENT "pattern"; LIST1 [ constr; q_occurrences ] SEP ",";
    "in"; constr
  | IDENT "eval"; IDENT "fold"; LIST1 constr; "in"; constr
  | IDENT "eval"; IDENT "unfold";
    LIST1 [ q_reference; q_occurrences ] SEP ","; "in"; constr
  | IDENT "eval"; IDENT "lazy"; q_strategy_flag; "in"; constr
  | IDENT "eval"; IDENT "cbn"; q_strategy_flag; "in"; constr
  | IDENT "eval"; IDENT "cbv"; q_strategy_flag; "in"; constr
  | IDENT "eval"; IDENT "simpl"; q_strategy_flag;
    OPT [ constr; q_occurrences ]; "in"; constr
  | IDENT "eval"; IDENT "hnf"; "in"; constr
  | IDENT "eval"; IDENT "red"; "in"; constr
  | IDENT "native_compute"; OPT [ constr; q_occurrences ]; OPT q_clause
  | IDENT "vm_compute"; OPT [ constr; q_occurrences ]; OPT q_clause
  | IDENT "pattern"; LIST1 [ constr; q_occurrences ] SEP ","; OPT q_clause
  | IDENT "fold"; LIST1 constr; OPT q_clause
  | IDENT "unfold"; LIST1 [ q_reference; q_occurrences ] SEP ",";
    OPT q_clause
  | IDENT "lazy"; q_strategy_flag; OPT q_clause
  | IDENT "cbn"; q_strategy_flag; OPT q_clause
  | IDENT "cbv"; q_strategy_flag; OPT q_clause
  | IDENT "simpl"; q_strategy_flag; OPT [ constr; q_occurrences ];
    OPT q_clause
  | IDENT "hnf"; OPT q_clause
  | IDENT "red"; OPT q_clause
  | IDENT "inversion_clear"; q_destruction_arg;
    OPT [ IDENT "as"; q_intropattern ]; OPT [ IDENT "in"; LIST1 q_ident ]
  | IDENT "inversion"; q_destruction_arg; OPT [ IDENT "as"; q_intropattern ];
    OPT [ IDENT "in"; LIST1 q_ident ]
  | IDENT "simple"; IDENT "inversion"; q_destruction_arg;
    OPT [ IDENT "as"; q_intropattern ]; OPT [ IDENT "in"; LIST1 q_ident ]
  | IDENT "edestruct"; LIST1 q_induction_clause SEP ",";
    OPT [ IDENT "using"; constr; q_with_bindings ]
  | IDENT "destruct"; LIST1 q_induction_clause SEP ",";
    OPT [ IDENT "using"; constr; q_with_bindings ]
  | IDENT "generalize";
    LIST1 [ constr; q_occurrences; OPT [ IDENT "as"; q_ident ] ] SEP ","
  | IDENT "einduction"; LIST1 q_induction_clause SEP ",";
    OPT [ IDENT "using"; constr; q_with_bindings ]
  | IDENT "induction"; LIST1 q_induction_clause SEP ",";
    OPT [ IDENT "using"; constr; q_with_bindings ]
  | IDENT "eremember"; constr; OPT [ IDENT "as"; q_ident ];
    OPT [ IDENT "eqn"; IDENT ":"; q_intropattern ]; OPT q_clause
  | IDENT "remember"; constr; OPT [ IDENT "as"; q_ident ];
    OPT [ IDENT "eqn"; IDENT ":"; q_intropattern ]; OPT q_clause
  | IDENT "eenough"; q_assert
  | IDENT "enough"; q_assert
  | IDENT "eassert"; q_assert
  | IDENT "assert"; q_assert
  | IDENT "eset"; q_pose; OPT q_clause
  | IDENT "set"; q_pose; OPT q_clause
  | IDENT "epose"; q_pose
  | IDENT "pose"; q_pose
  | IDENT "apply"; LIST1 [ constr; q_with_bindings ] SEP ",";
    OPT [ IDENT "in"; q_ident; OPT [ IDENT "as"; q_intropattern ] ]
  | IDENT "eapply"; LIST1 [ constr; q_with_bindings ] SEP ",";
    OPT [ IDENT "in"; q_ident; OPT [ IDENT "as"; q_intropattern ] ]
  | IDENT "eelim"; constr; q_with_bindings;
    OPT [ IDENT "using"; constr; q_with_bindings ]
  | IDENT "elim"; constr; q_with_bindings;
    OPT [ IDENT "using"; constr; q_with_bindings ]
  | IDENT "specialize"; [ constr; q_with_bindings ];
    OPT [ IDENT "as"; q_intropattern ]
  | IDENT "econstructor"; ltac2_expr LEVEL "5"; q_with_bindings
  | IDENT "econstructor"
  | IDENT "constructor"; ltac2_expr LEVEL "5"; q_with_bindings
  | IDENT "constructor"
  | IDENT "eright"; q_with_bindings
  | IDENT "right"; q_with_bindings
  | IDENT "eleft"; q_with_bindings
  | IDENT "left"; q_with_bindings
  | IDENT "eexists"; LIST0 q_bindings SEP ","
  | "exists"; LIST0 q_bindings SEP ","
  | IDENT "esplit"; q_with_bindings
  | IDENT "split"; q_with_bindings
  | IDENT "eintros"; q_intropatterns
  | IDENT "intros"; q_intropatterns
  | IDENT "solve"; "["; LIST0 (ltac2_expr LEVEL "6") SEP "|"; "]"
  | IDENT "first"; "["; LIST0 (ltac2_expr LEVEL "6") SEP "|"; "]"
  | "match!"; IDENT "reverse"; IDENT "goal"; "with"; q_goal_matching; "end"
  | "match!"; IDENT "goal"; "with"; q_goal_matching; "end"
  | "match!"; ltac2_expr LEVEL "6"; "with"; q_constr_matching; "end"
  | "multi_match!"; IDENT "reverse"; IDENT "goal"; "with"; q_goal_matching;
    "end"
  | "multi_match!"; IDENT "goal"; "with"; q_goal_matching; "end"
  | "multi_match!"; ltac2_expr LEVEL "6"; "with"; q_constr_matching; "end"
  | "lazy_match!"; IDENT "reverse"; IDENT "goal"; "with"; q_goal_matching;
    "end"
  | "lazy_match!"; IDENT "goal"; "with"; q_goal_matching; "end"
  | "lazy_match!"; ltac2_expr LEVEL "6"; "with"; q_constr_matching; "end" ]
| "5" LEFTA
  [ "fun"; LIST1 input_fun; type_cast; "=>"; ltac2_expr LEVEL "6"
  | "let"; rec_flag; LIST1 let_clause SEP "with"; "in"; ltac2_expr LEVEL "6"
  | "match"; ltac2_expr LEVEL "5"; "with"; branches; "end"
  | "if"; ltac2_expr LEVEL "5"; "then"; ltac2_expr LEVEL "5"; "else";
    ltac2_expr LEVEL "5" ]
| "4" LEFTA
  [  ]
| "3" LEFTA
  [ SELF; ","; LIST1 NEXT SEP "," ]
| "2" RIGHTA
  [ SELF; "::"; SELF ]
| "1" LEFTA
  [ SELF; ".("; qualid; ")"; ":="; ltac2_expr LEVEL "5"
  | SELF; ".("; qualid; ")"
  | SELF; LIST1 (ltac2_expr LEVEL "0") ]
| "0" LEFTA
  [ "()"
  | "("; ")"
  | "("; SELF; ")"
  | "("; SELF; ":"; ltac2_type; ")"
  | "["; LIST0 (ltac2_expr LEVEL "5") SEP ";"; "]"
  | "{"; tac2rec_fieldexprs; "}"
  | tactic_atom ] ]

```
</details>

Since I built this locally, I'm going to merge it without waiting for CI, and I'll assume that the documentation and changelog entries are fine (they look fine to me)

**Comment by JasonGross** [comment]
Oh, hrm, I guess I should wait for an approval by @coq/doc-maintainers ?  The change is so small, though.

**Comment by ppedrot** [comment]
I don't think this qualifies for documentation review. You should probably wait for the refman CI to terminate but that's about it.

**Comment by JasonGross** [comment]
@coqbot merge now

File: doc/changelog/07-vernac-commands-and-options/14093-fix-14092.rst
Comment (jfehrle) [inline doc/changelog/07-vernac-commands-and-options/14093-fix-14092.rst:3]: It only adds the ability to print the `ltac2_expr` nonterminal.  It doesn't include many things such as the various `q_*` symbols.

While I don't object to the change, the concept of `Print Grammar` is quite broken.  It certainly doesn't help users that Ltac2 has slightly different syntax for many build-in tactics.

Comment (JasonGross) [reply doc/changelog/07-vernac-commands-and-options/14093-fix-14092.rst:3]: I think `Print Grammar` should be generalized to accept any grammar entry.  Maybe we should open an issue for this?

Comment (jfehrle) [reply doc/changelog/07-vernac-commands-and-options/14093-fix-14092.rst:3]: I wouldn't recommend that.  We've spent a lot of time cleaning up the grammar we present in the documentation--clearly the right thing to do for the documentation.  Print Grammar doesn't match this and it would not be simple to make it match.  The currently printable nonterminals are the ones that can be altered at runtime by notations.

Another issue with Ltac2 is that many of the built-in tactics are not defined in .v files so they are not available unless the user defines them.  This will be an ongoing maintenance problem.

**Pull request merged**
